checkbox-provider-dss/units/jobs.pxu

```diff
@@ -262,3 +262,232 @@ command:
+    echo "DSS has started successfully!"
+  }
+  remove_ipex() {
+    echo "Removing IPEX"
+    sudo microk8s.kubectl delete -n dss deployment.apps/notebook-pytorch
+    sudo microk8s.kubectl delete -n dss service/user-notebook-pytorch
+  }
+  launch_notebook pytorch ipex
+  echo "Starting ipex import test"
+  pod=$(kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}' | grep -o 'notebook-pytorch.*')
+  echo "Found PyTorch pod: ${pod}"
+  sudo microk8s kubectl -n dss exec ${pod} -- python3 -c "import intel_extension_for_pytorch as ipex; import torch; import jupyter"
+  if [ "$?" = 0 ]
+  then
+    echo "PASS: Found module"
+    remove_ipex
+    exit 0
+  else
+    >&2 echo "FAIL: Did not find IPEX python module"
+    remove_ipex
+    exit 1
+  fi
+
+id: ipex/ipex_gpu_avial
+category_id: dss-regress
+flags: simple
+requires: executable.name == 'microk8s'
+_summary: Check IPEX GPU Availability
+estimated_duration: 5m
+command:
+  set -e
+  TIMEOUT=1200s
+  DSS_INTEL_MANIFESTS=/tmp/data-science-stack/poc/intel/manifests/
+  launch_notebook() {
+    local fullname=$1
+    local app=$2
+    echo -e "\n\$ dss start ${app} notebook"
+    sudo microk8s kubectl apply -f ${DSS_INTEL_MANIFESTS}/notebook-${app}.yaml
+    echo -e "\n\$ dss ${app} manifest applied"
+    sudo microk8s.kubectl wait \
+      --for condition=available \
+      --timeout $TIMEOUT \
+      -n dss\
+      deployment \
+      -l app=user-notebook-${fullname} > /tmp/ipex_log.txt
+    echo -e "\n\$ dss status"
+    MLFLOW_IP=$(sudo microk8s.kubectl get svc \
+      -n dss \
+      -o jsonpath="{.spec.clusterIP}" \
+      mlflow)
+    NOTEBOOK_IP=$(sudo microk8s.kubectl get svc \
+      -n dss \
+      -o jsonpath="{.spec.clusterIP}" \
+      user-notebook-${fullname})
+    echo "DSS has started successfully!"
+  }
+  remove_ipex() {
+    echo "Removing IPEX"
+    sudo microk8s.kubectl delete -n dss deployment.apps/notebook-pytorch
+    sudo microk8s.kubectl delete -n dss service/user-notebook-pytorch
+  }
+  launch_notebook pytorch ipex
+  echo "Starting ipex import test"
+  kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}'
+  pod=$(kubectl get pods -n dss -o=jsonpath='{.items..metadata.name}' | grep -o 'notebook-pytorch.*')
+  echo "Found PyTorch pod: ${pod}"
+  gpu_grep_out=$(sudo microk8s kubectl -n dss exec ${pod} -- python3 -c '
+  import sys
+  import torch
+  import intel_extension_for_pytorch as ipex
+  print(torch.__version__)
+  print(ipex.__version__)
+  try:
+    [print(f"[{i}]: {torch.xpu.get_device_properties(i)}") for i in range(torch.xpu.device_count())];
+    sys.exit(0)
+  except Exception:
+    print("Encountered an error getting XPU device properties")
+    sys.exit(1)
+  ' | grep "dev_type=.gpu" 2>&1)
+  remove_ipex
+  if [[ -z ${gpu_grep_out} ]]; then
+    echo "FAIL: No GPU found"
+    exit 1
+  elif [[ $gpu_grep_out =~ "gpu" ]]; then
+    echo "PASS: GPU found"
+    exit 0
+  else
+    echo "FAIL: Unexpected result"
```

Comment:
Maybe print the result here to help us diagnose the issue.

Reply:
Dropped unexpected result section since it's basically dead code